xds/src/main/java/io/grpc/xds/XdsNameResolverProvider.java

```diff
@@ -41,11 +41,24 @@
 @Internal
 public final class XdsNameResolverProvider extends NameResolverProvider {
 
-  private static final String SCHEME = "xds";
+  private String scheme = "xds";
+  private Map<String, ?> bootstrapOverride;
+
+  /**
+   * A convenient method to create a {@link XdsNameResolverProvider} with custom scheme and
+   * bootstrap.
+   */
+  public static XdsNameResolverProvider createForTest(String scheme,
+                                               @Nullable Map<String, ?> bootstrapOverride) {
+    XdsNameResolverProvider provider = new XdsNameResolverProvider();
+    provider.scheme = checkNotNull(scheme, "scheme");
```

Comment:
Add a constructor? That'd allow the fields to be final. You'd need to make a no-arg constructor as well.